### PR TITLE
docs: add release notes for 0.35.0

### PR DESCRIPTION
**Pull request by corban-beaird**
Add release notes for 0.35.0.

### CODE REVIEW

File: docs/release-notes.rst [hunk 1]
```diff
@@ -6,6 +6,196 @@
...
+
+**Release Date:** August 01, 2024
+
+**Breaking Changes**
+
+-  **Cluster:** The ``kubernetes_namespace`` field in the resource pool configuration is no longer
+   supported. Users can now submit workloads to specific namespaces by binding workspaces to
+   namespaces using the CLI or API.
+
+-  Master Configuration: ``resource_manager.name`` field is replaced by
```
Comment (tara-det-ai) [inline docs/release-notes.rst:24]:
```suggestion
- Master Configuration: Replace ``resource_manager.name`` with
```

File: docs/release-notes.rst [hunk 2]
```diff
@@ -6,6 +6,196 @@
...
+            resource managers in Determined.
+      -  During upgrade, replace ``name`` with ``cluster_name`` in the ``resource_manager`` section
+            of your master configuration YAML.
+
+-  Tasks: The :ref:`historical usage <historical-cluster-usage-data>` CSV file has been updated. The
+   header row for slot-hours is now named ``slot_hours`` instead of ``gpu_hours`` to accurately
+   reflect the allocation time for resource pools including those without GPUs. In addition, a new
+   column, ``resource_pool``, has been added to provide the resource pool for each allocation.
+
+-  Cluster: ``resources.agent_label`` task option and agent config ``label`` option have been
```
Comment (tara-det-ai) [inline docs/release-notes.rst:40]:
```suggestion
- Cluster: The ``resources.agent_label`` task option and the ``label`` option in the agent config have been
```

File: docs/release-notes.rst [hunk 3]
```diff
@@ -6,6 +6,196 @@
...
+   removed. Beginning with 0.20.0 release, these options have been ignored. Please remove any
+   remaining references from configuration files and use ``resource_pool`` instead.
+
+**New Features**
+
+-  WebUI: Add a "Namespace Bindings" section to the Create and Edit Workspace modals.
+
+   -  Users can input a namespace for a Kubernetes cluster. If no namespace is specified, the
+      workspace will use the ``resource_manager.default_namespace`` from the master configuration
```
Comment (tara-det-ai) [inline docs/release-notes.rst:49]:
```suggestion
workspace will be bound to the ``resource_manager.default_namespace`` field in the master configuration
```

File: docs/release-notes.rst [hunk 4]
```diff
@@ -6,6 +6,196 @@
...
+      YAML or default to the Kubernetes "default" namespace.
```
Comment (tara-det-ai) [inline docs/release-notes.rst:50]:
```suggestion
        YAML or the default Kubernetes namespace.
```

File: docs/release-notes.rst [hunk 5]
```diff
@@ -6,6 +6,196 @@
...
+   workspace creation with ``det w create <workspace-name> --cluster-name <cluster-name>
+   --auto-create-namespace --resource-quota <resource-quota>``, or later with ``det w resource-quota
+   set <workspace-id> <quota> --cluster-name <cluster-name>``.
+
+-  Add command to delete namespace bindings with ``det w bindings delete <workspace-id>
+   --cluster-name <cluster-name>``.
+
+-  Add command to list bindings for a workspace with ``det w bindings list <workspace-name>``. The
+   ``--cluster-name`` field is required only for MultiRM setups.
+
```
Comment (tara-det-ai) [inline docs/release-notes.rst:77]:
```suggestion

For help with workspace-namespace bindings, visit :ref:`Manage Workspace-Namespace Bindings <k8s-resource-caps>`.
```

File: docs/release-notes.rst [hunk 6]
```diff
@@ -6,6 +6,196 @@
...
+-  WebUI: Allow users to add/remove hyperparameters when performing hyperparameter search.
```
Comment (tara-det-ai) [inline docs/release-notes.rst:78]:
```suggestion
-  WebUI: Enable users to add or remove hyperparameters during hyperparameter searches.
```

File: docs/release-notes.rst [hunk 7]
```diff
@@ -6,6 +6,196 @@
...
+
+-  **Cluster:** The ``namespace`` field in the Kubernetes Resource Manager configuration has been
+   deprecated. Add a new field, ``default_namespace``. This field serves as the default namespace
```
Comment (tara-det-ai) [inline docs/release-notes.rst:81]:
```suggestion
-  **Cluster:** The ``namespace`` field in the Kubernetes Resource Manager configuration has been
   deprecated and replaced by ``default_namespace``. This field serves as the default namespace
```

Comment (tara-det-ai) [inline docs/release-notes.rst:81]:
```suggestion
       deprecated and replaced by ``default_namespace``. This field serves as the default namespace
```

File: docs/release-notes.rst [hunk 8]
```diff
@@ -6,6 +6,196 @@
...
+   for deploying namespaced resources when the workspace associated with a workload is not bound to
+   a specific namespace. The master configuration will accept either ``namespace`` or
+   ``default_namespace`` fields; however, providing both fields will result in an error.
+
+-  CLI: Add a feature where Determined offers the users to delete a workspace namespace binding by
```
Comment (tara-det-ai) [inline docs/release-notes.rst:86]:
```suggestion
- CLI: Add a command to delete a workspace-namespace binding:
```

File: docs/release-notes.rst [hunk 9]
```diff
@@ -6,6 +6,196 @@
...
+   using the command ``det w bindings delete <workspace-id> --cluster-name <cluster-name>``. An
```
Comment (tara-det-ai) [inline docs/release-notes.rst:87]:
```suggestion
:``det w bindings delete <workspace-id> --cluster-name <cluster-name>``. An
```

File: docs/release-notes.rst [hunk 10]
```diff
@@ -6,6 +6,196 @@
...
+   error will be thrown if the user tries to delete a default binding.
```
Comment (tara-det-ai) [inline docs/release-notes.rst:88]:
```suggestion
error will occur if attempting to delete a default binding.
```

File: docs/release-notes.rst [hunk 11]
```diff
@@ -6,6 +6,196 @@
...
+
+-  API: Add a feature where Determined offers the users to delete a workspace namespace binding by
```
Comment (tara-det-ai) [inline docs/release-notes.rst:90]:
```suggestion
- API: Add an endpoint to delete a workspace-namespace binding:
```

File: docs/release-notes.rst [hunk 12]
```diff
@@ -6,6 +6,196 @@
...
+   using the api endpoint ``api/v1/workspaces/1/namespace-bindings`` which takes in the workspace ID
```
Comment (tara-det-ai) [inline docs/release-notes.rst:91]:
```suggestion
 ``api/v1/workspaces/1/namespace-bindings``. This endpoint takes the workspace ID
```

File: docs/release-notes.rst [hunk 13]
```diff
@@ -6,6 +6,196 @@
...
+   and string array of cluster names as parameters. An error will be thrown if the user tries to
```
Comment (tara-det-ai) [inline docs/release-notes.rst:92]:
```suggestion
and string array of cluster names as parameters. An error will occur if attempting to 
```

File: docs/release-notes.rst [hunk 14]
```diff
@@ -6,6 +6,196 @@
...
+   delete a default binding.
+
+-  WebUI: View Pachyderm data lineage
```
Comment (tara-det-ai) [inline docs/release-notes.rst:95]:
```suggestion
WebUI: View :ref:`Pachyderm<pachyderm-integration>` data lineage.

File: docs/release-notes.rst [hunk 15]
```diff
@@ -6,6 +6,196 @@
...
+
+-  Experiments with configured Pachyderm data integration now display a link to the Pachyderm repo
+   in the trial view page. The link is also available when viewing checkpoints derived from the
+   Pachyderm data.
+
+-  Experiments: Obfuscate subfields of ``data.secrets`` in the :ref:`experiment configuration
+   <experiment-config-data>`.
+
+-  WebUI: In the Experimental features, Flat Runs View is now "on" by default in the :ref:`WebUI
+   <web-ui-if>`\. Users can still toggle this feature "off".
```
Comment (tara-det-ai) [inline docs/release-notes.rst:105]:
```suggestion
<web-ui-if>`. Users can still toggle this feature "off".
```

File: docs/release-notes.rst [hunk 16]
```diff
@@ -6,6 +6,196 @@
...
+
+-  This update improves the ability to compare model performance between different trials, based on
+   user feedback that most Determined users run single-trial experiments.
+
+-  "Experiments" is now renamed to "searches" and "trials" is now renamed to "runs" for better
```
Comment (tara-det-ai) [inline docs/release-notes.rst:110]:
```suggestion
- "Experiments" are now called "searches", and "trials" are now called "runs" for better
```

File: docs/release-notes.rst [hunk 17]
```diff
@@ -6,6 +6,196 @@
...
+   clarity.
+
+-  The "experiment list" is renamed the "run list", showing all trials from experiments in the
```
Comment (tara-det-ai) [inline docs/release-notes.rst:113]:
```suggestion
- The "experiment list" is now called the "run list", showing all trials from experiments in the
```

File: docs/release-notes.rst [hunk 18]
```diff
@@ -6,6 +6,196 @@
...
+   project. It functions similarly to the previous new experiment list.
+
+-  Multi-trial experiments can be viewed in the new searches view, which allows for sorting,
+   filtering and navigating multi-trial experiments.
+
+-  When viewing a multi-trial experiment, a list of trials is displayed, allowing for sorting,
+   filtering and arbitrary comparison between trials.
+
+-  WebUI: Add resource allocation information to trial details page.
```
Comment (tara-det-ai) [inline docs/release-notes.rst:122]:
```suggestion
- WebUI: Add resource allocation information to the trial details page.
```

File: docs/release-notes.rst [hunk 19]
```diff
@@ -6,6 +6,196 @@
...
+
+-  CLI: Add a new command, ``det cmd describe COMMAND_ID`` to allow users to fetch the metadata of a
+   single command.
+
+-  WebUI: Allow users to continue a canceled or errored multi-trial experiment for searcher type
+   ``random`` or ``grid``.
+
+**Improvements**
+
+-  Switch default AWS instance type from ``m5.large`` to ``m6i.large``. This change does not affect
```
Comment (tara-det-ai) [inline docs/release-notes.rst:132]:
```suggestion
- Switch the default AWS instance type from ``m5.large`` to ``m6i.large``. This change enhances performance without affecting 
```

File: docs/release-notes.rst [hunk 20]
```diff
@@ -6,6 +6,196 @@
...
+   the cost, but improves performance.
```
Comment (tara-det-ai) [inline docs/release-notes.rst:133]:
```suggestion
the cost.
```

File: docs/release-notes.rst [hunk 21]
```diff
@@ -6,6 +6,196 @@
...
+-  WebUI: Redirect SSO users to the SSO provider's authentication URIs when their session token has expired,
```
Comment (tara-det-ai) [inline docs/release-notes.rst:134]:
```suggestion
- WebUI: In the enterprise edition, redirect SSO users to the SSO provider's authentication URIs when their session token has expired,
```

File: docs/release-notes.rst [hunk 22]
```diff
@@ -6,6 +6,196 @@
...
+
+In the enterprise edition, Slurm resource manager still supports Singularity, Podman, or Apptainer
+use. For detailed instructions, visit :ref:deploy-on-slurm-pbs.
+
+Kubernetes Scheduling: Support for the priority scheduler for Kubernetes Resource Managers is
+discontinued and may be removed in a future release due to limited usage. Users should transition to
+the default scheduler. Visit :ref:`Kubernetes Default Scheduler <kubernetes-default-scheduler>` for
+details.
+
+**Deprecation**
```
Comment (tara-det-ai) [inline docs/release-notes.rst:187]: duplicate section title

Comment (corban-beaird) [reply docs/release-notes.rst:188]: Will remove! Thanks for catching this! I'll make sure any other duplicates are also handled

Comment (corban-beaird) [reply docs/release-notes.rst]: Cleaned up 😄

**Review by tara-det-ai — APPROVED** [review]
LGTM

File: docs/release-notes.rst [hunk 23]
```diff
@@ -6,6 +6,177 @@
...
+   supported. Users can now submit workloads to specific namespaces by binding workspaces to
+   namespaces using the CLI or API.
+
+-  Cluster: The ``resources.agent_label`` task option and the ``label`` option in the agent config
+   have been removed. Beginning with 0.20.0 release, these options have been ignored. Please remove
+   any remaining references from configuration files and use ``resource_pool`` instead.
+
+**New Features**
+
+-  WebUI/CLI/API: Allow admins to manage resource quotas for auto-created namespaces directly.
```
Comment (salonig23) [inline docs/release-notes.rst:45]: Allow admins to bind namespaces to workspaces and manage resource quotas for auto-created namespaces directly.

File: docs/release-notes.rst [hunk 24]
```diff
@@ -6,6 +6,177 @@
...
+
+   -  WebUI: Add a "Namespace Bindings" section to the Create and Edit Workspace modals.
+
+      -  Users can input a namespace for a Kubernetes cluster. If no namespace is specified, the
+         workspace will be bound to the ``resource_manager.default_namespace`` field in the master
+         configuration YAML or the default Kubernetes namespace.
```
Comment (salonig23) [inline docs/release-notes.rst:51]: ...configuration YAML or the "default" Kubernetes namespace...

File: docs/release-notes.rst (hunk 23 above)
Comment (corban-beaird) [reply docs/release-notes.rst:45]:
```suggestion
-  WebUI/CLI/API: Allow admins to bind namespaces to workspaces and manage resource quotas for auto-created namespaces directly.
```

File: docs/release-notes.rst [hunk 25]
```diff
@@ -6,6 +6,177 @@
...
+
+      -  Allow creating namespace bindings during workspace creation with ``det w create
+         <workspace-id> --namespace <namespace-name>`` or later with ``det w bindings set
+         <workspace-id> --namespace <namespace-name>``.
+
+      -  In the enterprise edition, users can use additional arguments ``--auto-create-namespace``
+         and ``--auto-create-namespace-all-clusters`` for auto-created namespaces. Set resource
+         quotas during workspace creation with ``det w create <workspace-name> --cluster-name
+         <cluster-name> --auto-create-namespace --resource-quota <resource-quota>``, or later with
+         ``det w resource-quota set <workspace-id> <quota> --cluster-name <cluster-name>``.
```
Comment (salonig23) [inline docs/release-notes.rst:74]: ...`--auto-create-namespace-all-clusters`` to bind workspaces to
            auto-created namespaces. Users can set resource quotas during workspace creation with ``det w
            create <workspace-name> --cluster-name <cluster-name> --auto-create-namespace
            --resource-quota <resource-quota>``, or later with ``det w resource-quota set
            <workspace-id> <quota> --cluster-name <cluster-name>`` if their workspace is bound to an auto-created namespace...

File: docs/release-notes.rst [hunk 26]
```diff
@@ -6,6 +6,177 @@
...
+
+      -  Add a command to delete namespace bindings with ``det w bindings delete <workspace-id>
+         --cluster-name <cluster-name>``.
+
+      -  Add a command to list bindings for a workspace with ``det w bindings list
+         <workspace-name>``. The ``--cluster-name`` field is required only for MultiRM setups.
```
Comment (salonig23) [inline docs/release-notes.rst:80]: Can we make this sentence its own bullet point and change it to:
The ``--cluster-name`` field is required only for MultiRM setups (when --auto-create-namespace-all-clusters is omitted)

File: docs/release-notes.rst [hunk 27]
```diff
@@ -6,6 +6,177 @@
...
+
+   -  API: Add new endpoints for creating and managing workspace namespace bindings.
+
+      -  Add POST and DELETE endpoints to ``/api/v1/workspaces/{workspace_id}/namespace-bindings``
+         for setting and deleting workspace namespace bindings.
+      -  Add a GET endpoint ``/api/v1/workspaces/{id}/list-namespace-bindings`` to list namespace
+         bindings for a workspace.
+      -  Add a POST endpoint ``/api/v1/workspaces/{id}/set-resource-quota`` to set resource quotas
+         on workspaces bound to auto-created namespaces.
+      -  Add a GET endpoint ``/api/v1/workspaces/{id}/get-k8-resource-quotas`` to retrieve enforced
```
Comment (salonig23) [inline docs/release-notes.rst:90]: Add a GET endpoint ``/api/v1/workspaces/{id}/get-k8s-resource-quotas`` to retrieve enforced Kubernetes resource quotas for bound namespaces

File: docs/release-notes.rst [hunk 28]
```diff
@@ -6,6 +6,177 @@
...
+
+**New Features**
+
+-  WebUI/CLI/API: Allow admins to bind namespaces to workspaces and manage resource quotas for auto-created namespaces directly.
+
+   -  WebUI: Add a "Namespace Bindings" section to the Create and Edit Workspace modals.
+
+      -  Users can input a namespace for a Kubernetes cluster. If no namespace is specified, the
+         workspace will be bound to the ``resource_manager.default_namespace`` field in the master
+         configuration YAML or the default Kubernetes namespace.
```
Comment (salonig23) [inline docs/release-notes.rst:51]:
```suggestion
         configuration YAML or the "default" Kubernetes namespace.
```

File: docs/release-notes.rst [hunk 29]
```diff
@@ -6,6 +6,177 @@
...
+
+      -  Allow creating namespace bindings during workspace creation with ``det w create
+         <workspace-id> --namespace <namespace-name>`` or later with ``det w bindings set
+         <workspace-id> --namespace <namespace-name>``.
+
+      -  In the enterprise edition, users can use additional arguments ``--auto-create-namespace``
+         and ``--auto-create-namespace-all-clusters`` for auto-created namespaces. Set resource
+         quotas during workspace creation with ``det w create <workspace-name> --cluster-name
+         <cluster-name> --auto-create-namespace --resource-quota <resource-quota>``, or later with
+         ``det w resource-quota set <workspace-id> <quota> --cluster-name <cluster-name>``.
```
Comment (salonig23) [inline docs/release-notes.rst:74]:
```suggestion
      -  In the enterprise edition, users can use additional arguments
            ``--auto-create-namespace`` and ``--auto-create-namespace-all-clusters`` to bind workspaces to
            auto-created namespaces. Users can set resource quotas during workspace creation with ``det w
            create <workspace-name> --cluster-name <cluster-name> --auto-create-namespace
            --resource-quota <resource-quota>``, or later with ``det w resource-quota set
            <workspace-id> <quota> --cluster-name <cluster-name>`` if their workspace is bound to an auto-created namespace.
```

File: docs/release-notes.rst [hunk 30]
```diff
@@ -6,6 +6,177 @@
...
+
+   -  API: Add new endpoints for creating and managing workspace namespace bindings.
+
+      -  Add POST and DELETE endpoints to ``/api/v1/workspaces/{workspace_id}/namespace-bindings``
+         for setting and deleting workspace namespace bindings.
+      -  Add a GET endpoint ``/api/v1/workspaces/{id}/list-namespace-bindings`` to list namespace
+         bindings for a workspace.
+      -  Add a POST endpoint ``/api/v1/workspaces/{id}/set-resource-quota`` to set resource quotas
+         on workspaces bound to auto-created namespaces.
+      -  Add a GET endpoint ``/api/v1/workspaces/{id}/get-k8-resource-quotas`` to retrieve enforced
```
Comment (salonig23) [inline docs/release-notes.rst:90]:
```suggestion
      -  Add a GET endpoint /api/v1/workspaces/{id}/get-k8s-resource-quotas to retrieve enforced Kubernetes resource quotas for workspace bound namespaces
```

File: docs/release-notes.rst [hunk 31]
```diff
@@ -6,6 +6,177 @@
...
+      -  Allow creating namespace bindings during workspace creation with ``det w create
+         <workspace-id> --namespace <namespace-name>`` or later with ``det w bindings set
+         <workspace-id> --namespace <namespace-name>``.
+
+      -  In the enterprise edition, users can use additional arguments ``--auto-create-namespace``
+         and ``--auto-create-namespace-all-clusters`` for auto-created namespaces. Set resource
+         quotas during workspace creation with ``det w create <workspace-name> --cluster-name
+         <cluster-name> --auto-create-namespace --resource-quota <resource-quota>``, or later with
+         ``det w resource-quota set <workspace-id> <quota> --cluster-name <cluster-name>``.
+
```
Comment (salonig23) [inline docs/release-notes.rst:75]:
```suggestion
      -  In the enterprise edition, users can use additional arguments ``--auto-create-namespace``
         and ``--auto-create-namespace-all-clusters`` to bind workspaces to auto-created namespaces. Users can set resource
         quotas during workspace creation with ``det w create <workspace-name> --cluster-name
         <cluster-name> --auto-create-namespace --resource-quota <resource-quota>``, or later with
         ``det w resource-quota set <workspace-id> <quota> --cluster-name <cluster-name>`` if their workspace is bound to an auto-created namespace.

```

File: docs/release-notes.rst [hunk 32]
```diff
@@ -6,6 +6,185 @@
...
 ###############
 
+**************
+ Version 0.35
+**************
+
+Version 0.35.0
+==============
+
+**Release Date:** August 02, 2024
```
Comment (tara-det-ai) [inline docs/release-notes.rst:16]: @corban-beaird we might need to change the data again

File: docs/release-notes.rst [hunk 33]
```diff
@@ -6,6 +6,185 @@
...
+**New Features**
+
+-  WebUI/CLI/API: Allow admins to bind namespaces to workspaces and manage resource quotas for
+   auto-created namespaces directly.
+
+   -  WebUI: Add a "Namespace Bindings" section to the Create and Edit Workspace modals.
+
+      -  Users can input a namespace for a Kubernetes cluster. If no namespace is specified, the
+         workspace will be bound to the ``resource_manager.default_namespace`` field in the master
+         configuration YAML or the "default" Kubernetes namespace.
```
Comment (salonig23) [inline docs/release-notes.rst:52]:
```suggestion
      -  Users can input a namespace for a Kubernetes cluster. If no namespace is specified, the
         workspace will be bound to the ``resource_manager.default_namespace`` field in the master
         configuration YAML when set or the release namespace. We fall back to the default Kubernetes namespace, "default", during non-helm determined deployments.
```

File: docs/release-notes.rst [hunk 34]
```diff
@@ -6,6 +6,185 @@
...
+-  WebUI: Add resource allocation information to the trial details page.
+
+-  WebUI: Allow users to continue a canceled or errored multi-trial experiment for searcher type
+   ``random`` or ``grid``.
+
+-  Cluster: The ``namespace`` field in the Kubernetes Resource Manager configuration has been
+   deprecated and replaced by ``default_namespace``. This field serves as the default namespace for
+   deploying namespaced resources when the workspace associated with a workload is not bound to a
+   specific namespace. The master configuration will accept either ``namespace`` or
+   ``default_namespace`` fields; however, providing both fields will result in an error.
```
Comment (salonig23) [inline docs/release-notes.rst:129]: delete this section from here (adding with changes to breaking changes)

File: docs/release-notes.rst [hunk 35]
```diff
@@ -6,6 +6,185 @@
...
+**************
+
+Version 0.35.0
+==============
+
+**Release Date:** August 05, 2024
+
+**Breaking Changes**
+
+-  Master Configuration: Replace ``resource_manager.name`` with ``resource_manager.cluster_name``
```
Comment (salonig23) [inline docs/release-notes.rst:20]: Add here:
Replace ``resource_manager.namespace`` with ``resource_manager.default_namespace``
The ``namespace`` field in the Kubernetes Resource Manager configuration is no longer supported and is replaced by ``default_namespace``. This field serves as the default namespace for deploying namespaced resources when the workspace associated with a workload is not bound to a specific namespace. If unset, the workloads will be sent to the release namespace, or the default Kubernetes namespace, "default", during non-helm determined deployments.

Comment (tara-det-ai) [reply docs/release-notes.rst:20]: @salonig23 to clarify, do you want to replace the following:

- Master Configuration: Replace ``resource_manager.name`` with ``resource_manager.cluster_name`` for better clarity and to support multiple resource managers.

with the following?

- Master Configuration: Replace resource_manager.namespace with resource_manager.default_namespace
The namespace field in the Kubernetes Resource Manager configuration is no longer supported and is replaced by default_namespace. This field serves as the default namespace for deploying namespaced resources when the workspace associated with a workload is not bound to a specific namespace. If unset, the workloads will be sent to the release namespace, or the default Kubernetes namespace, "default", during non-helm determined deployments.

or add it as a second "Master Configuration" item, like this?

- Master Configuration: Replace ``resource_manager.name`` with ``resource_manager.cluster_name`` for better clarity and to support multiple resource managers.

- Master Configuration: Replace ``resource_manager.namespace`` with ``resource_manager.default_namespace``.
The namespace field in the Kubernetes Resource Manager configuration is no longer supported and is replaced by ``default_namespace``. This field serves as the default namespace for deploying namespaced resources when the workspace associated with a workload is not bound to a specific namespace. If unset, the workloads will be sent to the release namespace, or the default Kubernetes namespace, "default", during non-helm determined deployments.

File: docs/release-notes.rst [hunk 36]
```diff
@@ -6,6 +6,190 @@
...
+   for better clarity and to support multiple resource managers.
+
+   -  Resource managers operate relative to the specific cluster providing resources for Determined
+      tasks, so changing the cluster will affect the resource manager's responses.
+   -  The ``cluster_name`` must be unique for all resource managers when deploying multiple resource
+      managers in Determined.
+   -  During upgrade, replace ``name`` with ``cluster_name`` in the ``resource_manager`` section of
```
Comment (amandavialva01) [inline docs/release-notes.rst:27]: When upgrading, specify a `resourceManager.clusterName` in your `values.yaml` to override `resource_manager.name` and/or remove the `name` field from your `resource_manager` config altogether.
 
- For additional resource managers, you must change `additional_resource_manager[i].name` to `additional_resource_manager[i].cluster_name` in your `values.yaml`.

File: docs/release-notes.rst (hunk 35 above)
Comment (amandavialva01) [reply docs/release-notes.rst:20]: The latter!

File: docs/release-notes.rst [hunk 37]
```diff
@@ -6,6 +6,190 @@
...
+
+-  Master Configuration: Replace ``resource_manager.namespace`` with
+   ``resource_manager.default_namespace``.
+
+   -  The namespace field in the Kubernetes Resource Manager configuration is no longer supported
+      and is replaced by ``default_namespace``.
+   -  This field serves as the default namespace for deploying namespaced resources when the
+      workspace associated with a workload is not bound to a specific namespace.
+   -  If unset, the workloads will be sent to the release namespace, or the default Kubernetes
+      namespace, "default", during non-helm determined deployments.
```
Comment (amandavialva01) [inline docs/release-notes.rst:38]: If unset, the workloads will be sent to the release namespace during determined helm installs or upgrades and will be sent to the default Kubernetes namespace, "default", during non-helm determined deployments.

File: docs/release-notes.rst [hunk 38]
```diff
@@ -6,6 +6,190 @@
...
+
+-  Tasks: The :ref:`historical usage <historical-cluster-usage-data>` CSV file has been updated. The
+   header row for slot-hours is now named ``slot_hours`` instead of ``gpu_hours`` to accurately
+   reflect the allocation time for resource pools including those without GPUs. In addition, a new
+   column, ``resource_pool``, has been added to provide the resource pool for each allocation.
+
+-  Cluster: The ``kubernetes_namespace`` field in the resource pool configuration is no longer
+   supported. Users can now submit workloads to specific namespaces by binding workspaces to
+   namespaces using the CLI or API.
```
Comment (amandavialva01) [inline docs/release-notes.rst:47]: using the CLI, API, or the webUI

File: docs/release-notes.rst [hunk 39]
```diff
@@ -6,6 +6,190 @@
...
+
+-  Master Configuration: Replace ``resource_manager.name`` with ``resource_manager.cluster_name``
+   for better clarity and to support multiple resource managers.
+
+   -  Resource managers operate relative to the specific cluster providing resources for Determined
+      tasks, so changing the cluster will affect the resource manager's responses.
+   -  The ``cluster_name`` must be unique for all resource managers when deploying multiple resource
+      managers in Determined.
+   -  During upgrade, replace ``name`` with ``cluster_name`` in the ``resource_manager`` section of
+      your master configuration YAML.
```
Comment (amandavialva01) [inline docs/release-notes.rst:28]:
```suggestion
   -  When upgrading, specify `resourceManager.clusterName` in your `values.yaml` to override `resource_manager.name` and/or remove the `name` field from your `resource_manager` config altogether.
- For additional resource managers, you must change `additional_resource_manager[i].name` to `additional_resource_manager[i].cluster_name` in your `values.yaml`.
```

File: docs/release-notes.rst (hunk 38 above)
Comment (amandavialva01) [inline docs/release-notes.rst:47]:
```suggestion
   namespaces using the CLI, WebUI, or API.
```

File: docs/release-notes.rst (hunk 37 above)
Comment (amandavialva01) [inline docs/release-notes.rst:38]:
```suggestion
   -  If unset, the workloads will be sent to the release namespace during determined helm installs or upgrades and will be sent to the default Kubernetes namespace, "default", during non-helm determined deployments.
```

File: docs/release-notes.rst [hunk 40]
```diff
@@ -6,6 +6,189 @@
...
+
+-  Master Configuration: Replace ``resource_manager.namespace`` with
+   ``resource_manager.default_namespace``.
+
+   -  The namespace field in the Kubernetes Resource Manager configuration is no longer supported
+      and is replaced by ``default_namespace``.
+   -  This field serves as the default namespace for deploying namespaced resources when the
+      workspace associated with a workload is not bound to a specific namespace.
+   -  If unset, the workloads will be sent to the release namespace during determined helm installs or upgrades and will be sent to the default Kubernetes namespace, "default", during non-helm determined deployments.
+
```
Comment (tara-det-ai) [inline docs/release-notes.rst:38]:
```suggestion

-  Master Configuration: Replace ``resource_manager.namespace`` with ``resource_manager.default_namespace``.
   The namespace field in the Kubernetes Resource Manager configuration is no longer supported and is replaced 
   by ``default_namespace``. This field serves as the default namespace for deploying namespaced resources 
   when the workspace associated with a workload is not bound to a specific namespace. If unset, the workloads 
   will be sent to the release namespace, or the default Kubernetes namespace, "default", during 
   non-helm determined deployments.
```

File: docs/release-notes.rst (hunk 35 above)
Comment (tara-det-ai) [reply docs/release-notes.rst:20]: ok thanks!
@corban-beaird this item goes just above tasks. i've added a suggested edit. below. you can incorporated the suggested edit and then resolve these comments.

Comment (corban-beaird) [reply docs/release-notes.rst:20]: Awesome thanks Tara! 😄

File: docs/release-notes.rst [hunk 41]
```diff
@@ -6,6 +6,195 @@
...
 ###############
 
+**************
+ Version 0.35
+**************
+
+Version 0.35.0
+==============
+
+**Release Date:** August 06, 2024
```
Comment (tara-det-ai) [inline docs/release-notes.rst:16]: August 9th?

File: docs/release-notes.rst [hunk 42]
```diff
@@ -6,6 +6,195 @@
...
+         --cluster-name <cluster-name>`` if their workspace is bound to an auto-created namespace.
+
+      -  Add a command to delete namespace bindings with ``det w bindings delete <workspace-id>
+         --cluster-name <cluster-name>``.
+
+      -  Add a command to list bindings for a workspace with ``det w bindings list
+         <workspace-name>``.
+
+      -  The ``--cluster-name`` field is required only for MultiRM setups when
+         --auto-create-namespace-all-clusters is omitted.
```
Comment (tara-det-ai) [inline docs/release-notes.rst:100]:
```suggestion
``--auto-create-namespace-all-clusters`` is omitted.
```

File: docs/release-notes.rst (hunk 41 above)
Comment (corban-beaird) [reply docs/release-notes.rst:16]: probably a safe bet, worst case scenario; I'll bump it back to the 8th tomorrow if things wrap up 😄

**Review by hkumar92 — APPROVED** [review]

**Review by tara-det-ai — APPROVED** [review]

**Pull request merged**